### ISSUE TITLE
Prediction with partial observations

### ISSUE THREAD

**Issue by zhh210** [GAMES-UChile/mogptk#14]
It's a great feature to allow missing values (e.g., remove_range) in the training. I'm wondering if it is supported in prediction/inference? For example, in prediction I may have certain channel values available but other channels to be predicted. Thanks.

**Comment by tdewolff** [comment]
You could set prediction ranges per channel (https://games-uchile.github.io/mogptk/data.html#mogptk.data.Data.set_prediction_range) and you could also define a custom prediction `x`, i.e. the x-values for which to predict (https://games-uchile.github.io/mogptk/data.html#mogptk.data.Data.set_prediction_x). I think that should give you the tools you need, unless I misunderstand?

**Comment by zhh210** [comment]
Thanks @tdewolff , Let me rephrase this way: I'm able to train a model on the range of 2019-01-01 to 2019-12-31 and am using it to forecasting 2020. As time goes, newer observations of 2020 are kept being added, can I use use the trained model (on 2019 data) with observations of 2020-01-01 to predict 2020-01-02 (without retraining the model)?

**Comment by Ale-Cuevas** [comment]
Currently there is no straightforward way to "update" a model with new data, as the `dataset` atribute must be changed and then the internal gpflow model rebuilt. 
A solution is to save the parameters of the trained model and create a another model with the newer observations, then load the parameters from the previous model.
An example with the MOSM kernel would be:
```
model.save_parameters("trained_model")

new_model = mogptk.MOSM(new_dataset)
new_model.load_parameters("trained_model")
```

**Comment by zhh210** [comment]
@Ale-Cuevas Thanks for pointing out. Yes I believe this workaround should work. When I was trying to replicate the saved model:
```
mosm = mogptk.MOSM(dataset, Q=3)
mosm.init_parameters('BNSE')
mosm.predict()
mosm.plot_prediction()
mosm.save_parameters('trained_model')
```
The saved model once re-loaded doesn't give same predictions even on the same dataset:
```
mosm2 = mogptk.MOSM( dataset, Q=3)
mosm2.load_parameters("trained_model")
mosm2.predict()
mosm2.plot_prediction()
```
or am I missing something?

**Comment by zhh210** [comment]
Ok, I found it. The lines in [https://games-uchile.github.io/mogptk/model.html#mogptk.model.model.set_parameter](https://games-uchile.github.io/mogptk/model.html#mogptk.model.model.set_parameter):
      ```
if v < gpflow.config.default_positive_minimum():
            val[i] = gpflow.config.default_positive_minimum()
```
Changed the mean value when calling load_parameters()
